### ISSUE TITLE
DevMode Emunand

### ISSUE THREAD

**Issue by fox8091** [roxas75/rxTools#222]
Would it be possible to implement devmode for emunand?

**Comment by motezazer** [comment]
Why would you need that?

**Comment by hartmannaf** [comment]
Maybe he wants wo get clean memory dumps of processes of newer firmware versions.
Or to test other patches to do things implemented patches are doing.
Its something that could only be helpful for patch devs I think, so it would be a good nice to have feature, but its not needed for the most people.

**Comment by fox8091** [comment]
How hard would this be to do?

**Comment by fox8091** [comment]
And I'm not using it for patch dev. I just need to load it without firmlaunch. To just use the firm in the emunand.

**Comment by AlbertoSONIC** [comment]
By firmlaunching, every rxTools user has 9.5 (n3ds) or 9.9 firm (o3ds) launched. So we just need to create patches for 2 firms. If we support emunand, then we would have to adjust those patches for every single firmware, from 4.1 to 9.2... Which is crazy.

**Comment by hartmannaf** [comment]
why do you want to launch the emunand firm? What would be your benefit?

**Comment by AlbertoSONIC** [comment]
We don't launch the emunand firmware. We launch the .bin file from rxtools/data/, which is 9.5 firm on n3ds and 9.9 on the o3ds. We don't care what's the emunand version, as our patches are done on the firm, not on the emunand. All this means that we have to create 2 seta of patches, one for 9.5 n3ds firm and one for 9.9 o3ds firm. Devmode + emunand means firmlaunching the firm contained in the sysnand, then applying emunand patches, which would mean to have a patch for evert firm, from 4.1 to 9.2..

**Comment by hartmannaf** [comment]
@AlbertoSONIC I know, I should have written @fox8091.
I thought about something like emunand with only the needed patches, so that there won't be a rebbot fix patch for ram expended games, and no auto arm11 patches. So that we have mostly clean emunand ram, to test and develop patches for arm 11.

But I don't know why @fox8091 would like to use the emunand firm and not the newest possible like we do at the moment. That's why I tried to ask him what the benefits would be.

**Comment by fox8091** [comment]
I'm testing something in 2.0.0-2E.

**Comment by Syphurith** [comment]
@fox8091 I've recalled something. When rxTools itself came to the 2.4 official release, there was no sig-patch or firmlaunch for both NANDs. That was only after *someone* patched it to let it be a real cfw.
Maybe you could just find an 2.4 official release somewhere else, for example wololo.
If you could find it nowhere, you might go to twitter to ask roxas75 himself for one.
Note: I can not ensure that would work with the 2.0.0 console, since the original runs on 4.x MSET. And for your case i think the HACKER @Normmatt could be a better expert. (sorry for disturbing you..)

BTW.. @mid-kid @dukesrg @motezazer  I'm wondering why FBI injection could work, now. I don't know how NATIVE_FIRM package differents from the H&S App and NATIVE_FIRM. If there is no severe problem to inject a patched FIRM.. Well i might still be dreaming hard. Could you please tell me?
Note: If you want to test it out, you would have to know how to make that .app and .tmd file. I'm afraid it is more likely to fail. Also you must have yours hard-modded before ever testing.

**Comment by mid-kid** [comment]
First of all, rxTools 2.4 DID do what they now call "firmlaunch". How else would it achieve emuNAND?
Th H&S app can be replaced, because it's not critical for booting, and we have enough ways of disabling the checks before it is run (notice how it won't start without signature patches?). However, the NATIVE_FIRM is different in that (aside from being installed in a separate partition) it is one of the critical pieces needed to boot, and it's checked by the bootloader, a check we can't disable.

P.s. don't know if someone already replied before this message got sent.

On 29 agost de 2015 7:30:53 CEST, Syphurith <notifications@github.com> wrote:
>@fox8091 I've recalled something. When rxTools itself came to the 2.4
>official release, there was no sig-patch or firmlaunch for both NANDs.
>That was only after *someone* patched it to let it be a real cfw.
>Maybe you could just find an 2.4 official release somewhere else, for
>example wololo.
>If you could find it nowhere, you might go to twitter to ask roxas75
>himself for one.
>Note: I can not ensure that would work with the 2.0.0 console, since
>the original runs on 4.x MSET. And for your case i think the HACKER
>@Normmatt could be a better expert. (sorry for disturbing you..)
>
>BTW.. @mid-kid @dukesrg @motezazer  I'm wondering why FBI injection
>could work, now. I don't know how NATIVE_FIRM package differents from
>the H&S App and NATIVE_FIRM. If there is no severe problem to inject a
>patched FIRM.. Well i might still be dreaming hard. Could you please
>tell me?
>Note: If you want to test it out, you would have to know how to make
>that .app and .tmd file. I'm afraid it is more likely to fail. Also you
>must have yours hard-modded before ever testing.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/roxas75/rxTools/issues/222#issuecomment-135943285

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by mid-kid** [comment]
First of all, rxTools 2.4 DID do what they now call "firmlaunch". How else would it achieve emuNAND?
Th H&S app can be replaced, because it's not critical for booting, and we have enough ways of disabling the checks before it is run (notice how it won't start without signature patches?). However, the NATIVE_FIRM is different in that (aside from being installed in a separate partition) it is one of the critical pieces needed to boot, and it's checked by the bootloader, a check we can't disable.

P.s. don't know if someone already replied before this message got sent.

On 29 agost de 2015 7:30:53 CEST, Syphurith <notifications@github.com> wrote:
>@fox8091 I've recalled something. When rxTools itself came to the 2.4
>official release, there was no sig-patch or firmlaunch for both NANDs.
>That was only after *someone* patched it to let it be a real cfw.
>Maybe you could just find an 2.4 official release somewhere else, for
>example wololo.
>If you could find it nowhere, you might go to twitter to ask roxas75
>himself for one.
>Note: I can not ensure that would work with the 2.0.0 console, since
>the original runs on 4.x MSET. And for your case i think the HACKER
>@Normmatt could be a better expert. (sorry for disturbing you..)
>
>BTW.. @mid-kid @dukesrg @motezazer  I'm wondering why FBI injection
>could work, now. I don't know how NATIVE_FIRM package differents from
>the H&S App and NATIVE_FIRM. If there is no severe problem to inject a
>patched FIRM.. Well i might still be dreaming hard. Could you please
>tell me?
>Note: If you want to test it out, you would have to know how to make
>that .app and .tmd file. I'm afraid it is more likely to fail. Also you
>must have yours hard-modded before ever testing.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/roxas75/rxTools/issues/222#issuecomment-135943285

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Syphurith** [comment]
@mid-kid Thanks for your reply. I understand it better now, thanks.

**Comment by fox8091** [comment]
Doesn't devmode already have to have patches for every firmware? In that case, couldn't you just use them?

**Comment by kot2002** [comment]
@fox8091 look into CakesFW source code mid-kid@CakesForeveryWan

**Comment by urherenow** [comment]
Launching emunand without firmlaunch is useful for people who want a clean upgrade to 9.5, for example. Any other way is not a proper upgrade since your console would brick if you flashed it to sysnand. NATIVE_FIRM and TWL_FIRM don't get updated when firmlaunchax is used.

**Comment by AlbertoSONIC** [comment]
@urherenow It's impossible to launch emunand without firmlaunch. Also, I think this "devmode emunand" won't happen anytime soon so I guess we can close this (for now) ;)
